docs/index.md

```diff
@@ -61,18 +61,72 @@ your applications.
 <a name="protocolbuffers"></a>
 By default gRPC uses *protocol buffers*, Google’s
 mature open source mechanism for serializing structured data (although it
-can be used with other data formats such as JSON). As you'll
-see in our example below, you define gRPC services using *proto files*,
-with method parameters and return types specified as protocol buffer message
-types. You
-can find out lots more about protocol buffers in the [Protocol Buffers
-documentation](https://developers.google.com/protocol-buffers/docs/overview).
+can be used with other data formats such as JSON). Here's a quick intro to how
+it works. If you're already familiar with protocol buffers, feel free to skip
+ahead to the next section.
+
+The first step when working with protocol buffers is to define the structure
+forthe data you want to serialize in a *proto file*: this is an ordinary text
+file with a `.proto` extension. Protocol buffer data is structured as
+*messages*, where each message is a small logical record of information
+containing a series of name-value pairs called *fields*. Here's a simple
+example:
+
+```
+message Person {
+  string name = 1;
+  int32 id = 2;
+  bool has_ponycopter = 3;
+}
+```
+
+Then, once you've specified your data structures, you use the protocol buffer
+compiler `protoc` to generate data access classes in your preferred language(s)
+from your proto definition. These provide simple accessors for each field
+(like `name()` and `set_name()`) as well as methods to serialize/parse
+the whole structure to/from raw bytes – so, for instance, if your chosen
+language is C++, running the compiler on the above example will generate a
+class called `Person`. You can then use this class in your application to
+populate, serialize, and retrieve Person protocol buffer messages.
+
+As you'll see in more detail in our examples, you define gRPC services 
+in ordinary proto files, with RPC method parameters and return types specified as
+protocol buffer messages:
+
+```
+// The greeter service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+
+With gRPC you also use `protoc` (this time with a special gRPC plugin) to
```

Comment:
I would change this to something like, 
gRPC uses the `protoc` (..) with a gRPC plugin.. 
The reason is that in some languages, users do not directly interact with the protoc plugin to generate code. For example, in java, the gradle or maven plugin does this for them, similarly for C#. For python, we have a python module that can be invoked. Therefore the language here might be better if it did not imply how the user would user the tool, but how the plugin and protoc tool work together to generate the code.